test/gcs/mock/mock_test.go

```diff
@@ -0,0 +1,1057 @@
+						t.Errorf("expected error %v", v.Err)
+					} else if err.Error() != v.Err.Error() {
+						t.Errorf("expected error %v, got error %v", v.Err, err)
+					}
+				case MethodAttrObject:
+					for i := uint8(0); i < v.NumCall; i++ {
+						mockClient.AttrObject(ctx, bkt, dirPath)
+					}
+
+					if _, err := mockClient.AttrObject(ctx, bkt, dirPath); err == nil {
+						t.Errorf("expected error %v", v.Err)
+					} else if err.Error() != v.Err.Error() {
+						t.Errorf("expected error %v, got error %v", v.Err, err)
+					}
+				case MethodCopyObject:
+					for i := uint8(0); i < v.NumCall; i++ {
+						mockClient.CopyObject(ctx, bkt, dirPath, bkt, dirPath)
+					}
+
+					if err := mockClient.CopyObject(ctx, bkt, dirPath, bkt, dirPath); err == nil {
+						t.Errorf("expected error %v", v.Err)
+					} else if err.Error() != v.Err.Error() {
+						t.Errorf("expected error %v, got error %v", v.Err, err)
+					}
+				case MethodReadObject:
+					for i := uint8(0); i < v.NumCall; i++ {
+						mockClient.ReadObject(ctx, bkt, dirPath)
+					}
+
+					if _, err := mockClient.ReadObject(ctx, bkt, dirPath); err == nil {
+						t.Errorf("expected error %v", v.Err)
+					} else if err.Error() != v.Err.Error() {
+						t.Errorf("expected error %v, got error %v", v.Err, err)
+					}
+				case MethodWriteObject:
+					for i := uint8(0); i < v.NumCall; i++ {
+						mockClient.WriteObject(ctx, bkt, dirPath, []byte{})
+					}
+
+					if _, err := mockClient.WriteObject(ctx, bkt, dirPath, []byte{}); err == nil {
+						t.Errorf("expected error %v", v.Err)
+					} else if err.Error() != v.Err.Error() {
+						t.Errorf("expected error %v, got error %v", v.Err, err)
+					}
+				case MethodDeleteObject:
+					for i := uint8(0); i < v.NumCall; i++ {
+						mockClient.DeleteObject(ctx, bkt, dirPath)
+					}
+
+					if err := mockClient.DeleteObject(ctx, bkt, dirPath); err == nil {
+						t.Errorf("expected error %v", v.Err)
+					} else if err.Error() != v.Err.Error() {
+						t.Errorf("expected error %v, got error %v", v.Err, err)
+					}
+				case MethodDownload:
+					for i := uint8(0); i < v.NumCall; i++ {
+						mockClient.Download(ctx, bkt, dirPath, dirPath)
+					}
+
+					if err := mockClient.Download(ctx, bkt, dirPath, dirPath); err == nil {
+						t.Errorf("expected error %v", v.Err)
+					} else if err.Error() != v.Err.Error() {
+						t.Errorf("expected error %v, got error %v", v.Err, err)
+					}
+				case MethodUpload:
+					for i := uint8(0); i < v.NumCall; i++ {
+						mockClient.Upload(ctx, bkt, dirPath, dirPath)
+					}
+
+					if err := mockClient.Upload(ctx, bkt, dirPath, dirPath); err == nil {
+						t.Errorf("expected error %v", v.Err)
+					} else if err.Error() != v.Err.Error() {
+						t.Errorf("expected error %v, got error %v", v.Err, err)
+					}
+				default:
+					t.Errorf("unknown method")
+				}
+			}
+		})
+	}
+}
+
+func TestClearError(t *testing.T) {
+	ctx := context.Background()
+	bkt := "dummy"
```

Comment:
const?

Reply:
I dont think you can mark a var constant in a function in golang. And it would be better to have the parameters of the test be in the test itself rather than somewhere outside. For future and for readability.